specs/lightclient/validator.md

```diff
@@ -0,0 +1,395 @@
+  - [Sync committees](#sync-committees)
+    - [Sync committee signatures](#sync-committee-signatures)
+      - [Prepare sync committee signature](#prepare-sync-committee-signature)
+      - [Broadcast sync committee signature](#broadcast-sync-committee-signature)
+    - [Sync committee contributions](#sync-committee-contributions)
+      - [Aggregation selection](#aggregation-selection)
+      - [Construct sync committee contribution](#construct-sync-committee-contribution)
+      - [Broadcast sync committee contribution](#broadcast-sync-committee-contribution)
+- [Sync committee subnet stability](#sync-committee-subnet-stability)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This document represents the expected behavior of an "honest validator" with respect to HF1 of the Ethereum 2.0 protocol. It builds on the [previous document for the behavior of an "honest validator" from Phase 0](../phase0/validator.md) of the Ethereum 2.0 protocol. This previous document is referred to below as the "Phase 0 document".
+
+HF1 introduces a new type of committee: the sync committee. Sync committees are responsible for signing each block of the canonical chain and there exists an efficient algorithm for light clients to sync the chain using the output of the sync committees. See the [sync protocol](./sync-protocol.md) for further details on the light client sync. Under this network upgrade, validators track their participation in this new committee type and produce the relevant signatures as required. Block proposers incorporate the (aggregated) sync committee signatures into each block they produce.
+
+## Prerequisites
+
+All terminology, constants, functions, and protocol mechanics defined in the [HF1 -- The Beacon Chain](./beacon-chain.md) doc are requisite for this document and used throughout. Please see this document before continuing and use as a reference throughout.
+
+## Custom types
+
+| Name | SSZ equivalent | Description |
+| - | - | - |
+| `SyncCommitteeIndex` | `uint64` | Index of a validator within a sync committee |
+
+## Constants
+
+### Misc
+
+| Name | Value | Unit |
+| - | - | :-: |
+| `TARGET_AGGREGATORS_PER_SYNC_COMMITTEE` | `2**2` (= 4) | validators |
+| `SYNC_COMMITTEE_SUBNET_COUNT` | `8` | The number of sync committee subnets used in the gossipsub aggregation protocol. |
+
+## Containers
+
+### SyncCommitteeSignature
+
+```python
+class SyncCommitteeSignature(Container):
+    # Slot to which this contribution pertains
+    slot: Slot
+    # Index in the sync committee corresponding to the validator that produced this signature
+    index: SyncCommitteeIndex
+    # Signature by the validator over the block root of `slot`
+    signature: BLSSignature
+```
+
+### SyncCommitteeContribution
+
+```python
+class SyncCommitteeContribution(Container):
+    # Slot to which this contribution pertains
+    slot: Slot
```

Comment:
Similarly, I think we'll want block_root here. but we can debate a bit more tomorrow